### PR TITLE
[SPARK-49989][PYTHON][CONNECT] Support kde/density plots

### PR DESCRIPTION
**Pull request by xinrong-meng**
### What changes were proposed in this pull request?
Support kde/density plots with plotly backend on both Spark Connect and Spark classic.

### Why are the changes needed?
While Pandas on Spark supports plotting, PySpark currently lacks this feature. The proposed API will enable users to generate visualizations. This will provide users with an intuitive, interactive way to explore and understand large datasets directly from PySpark DataFrames, streamlining the data analysis workflow in distributed environments.

See more at [PySpark Plotting API Specification](https://docs.google.com/document/d/1IjOEzC8zcetG86WDvqkereQPj_NGLNW7Bdu910g30Dg/edit?usp=sharing) in progress.

Part of https://issues.apache.org/jira/browse/SPARK-49530.

### Does this PR introduce _any_ user-facing change?
Yes. kde/density plots are supported as shown below.

```py
>>> data = [
...     (1.0, 4.0),
...     (2.0, 4.0),
...     (2.5, 4.5),
...     (3.0, 5.0),
...     (3.5, 5.5),
...     (4.0, 6.0),
...     (5.0, 6.0)
... ]
>>> columns = ["x", "y"]
>>> df = spark.createDataFrame(data, columns)
>>> fig1 = df.plot.kde(column=["x", "y"], bw_method=0.3, ind=100)
>>> fig1.show()
>>> fig2 = df.plot(kind="kde", column="x", bw_method=0.3, ind=20)
>>> fig2.show()
```

### How was this patch tested?
Unit tests.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

File: python/pyspark/sql/plot/core.py [hunk 1]
```diff
@@ -388,6 +396,127 @@ def box(
         """
         return self(kind="box", column=column, precision=precision, **kwargs)
 
+    def kde(
+        self,
+        column: Union[str, List[str]],
```
Comment (xinrong-meng) [inline python/pyspark/sql/plot/core.py:401]: Create https://issues.apache.org/jira/browse/SPARK-49999 for a follow-up on optional "column" parameter support in box, kde and hist plots

Comment (xinrong-meng) [reply python/pyspark/sql/plot/core.py:401]: Create https://issues.apache.org/jira/browse/SPARK-50000 for  optional "bw_method" in both Pandas on Spark and PySpark

**Comment by xinrong-meng** [comment]
@zhengruifeng @HyukjinKwon may I get a review please?

File: python/pyspark/sql/pandas/utils.py [hunk 2]
```diff
@@ -94,3 +94,33 @@ def require_minimum_pyarrow_version() -> None:
             errorClass="ARROW_LEGACY_IPC_FORMAT",
             messageParameters={},
         )
+
+
+def require_minimum_numpy_version() -> None:
+    """Raise ImportError if minimum version of NumPy is not installed"""
```
Comment (zhengruifeng) [inline python/pyspark/sql/pandas/utils.py:100]: numpy is a dependency of pandas, so it is available in connect;
but not a mandatary one in classic.

It seems `kde` actually only need `np.linspace`, whose logic is not complex, so we may remove the direct dependency of `numpy` later,

Comment (xinrong-meng) [reply python/pyspark/sql/pandas/utils.py:100]: Good point! Created https://issues.apache.org/jira/browse/SPARK-50020 as a followup.

**Review by zhengruifeng — APPROVED** [review]

**Comment by zhengruifeng** [comment]
merged to master

**Comment by xinrong-meng** [comment]
Thank you @zhengruifeng !

**Pull request closed without merging**